sycl/test-e2e/Matrix/SG32/joint_matrix_colA_rowB_colC.cpp

```diff
@@ -11,7 +11,7 @@
 // RUN: %{build} -o %t.out
 // RUN: %{run} %t.out
 
-// XFAIL:*
+// XFAIL: run-mode
```

Comment:
These changes are to differentiate between tests that fail when compiling vs tests that fail when running. These tests originally are able to build so marking as `XFAIL: run-mode` makes sure these tests arent reported as `XPASS` when building only.

Reply:
I want (very strongly) to start with explicit tests markup first before we introduce any heuristics in the lit logic, especially if that's really heuristics and not some bullet-proof correct logic.

Reply:
i guess im just worried this feature will slow down/confuse developers writing tests, im willing to proceed with this as-is and see what happens, but we should be ready to revert/fix the problem fast if some issue comes up

Reply:
> i guess im just worried this feature will slow down/confuse developers writing tests, im willing to proceed with this as-is and see what happens, but we should be ready to revert/fix the problem fast if some issue comes up

For now since there is no automatic running on CI enabled, we'll just have to monitor the failures manually. 

However once this is integrated into the post/pre commit I think it should be pretty straightforward to figure out the appropriate action given the ci's behaviour: 
- If we see that a test passes the build stage but fails the run stage we have a runfail, and if this is expected we would mark it as XFAIL/UNSUPPORTED on `run-mode`. 
- If it fails on both the build and run stage then we have a compfail and we would mark as XFAIL/UNSUPPORTED * (Or we could mark as XFAIL for a feature that is non-device-specific, like `linux`).